proto/tendermint/abci/types.proto

```diff
@@ -1,8 +1,6 @@
 syntax = "proto3";
 package tendermint.abci;
 
-option go_package = "github.com/tendermint/tendermint/abci/types";
```

Comment:
when this is imported into Tendermint, will it not cause package issues?

`package tendermint_abci`

Reply:
whoops, never mind, I read the other PR now. 

The only downside of doing it this way is, users of our photo files will need to append option go_package as well, this could cause confusion. Before it was plug and play.

Reply:
Unfortunately, this is really a downside of hosting the protos in the spec repo in general. If users were to try to generate these protos themselves using the old `option`, they would have created protos with import paths that do not match their local module name. As a result of this, they would have two options, both of which are not ideal.

Option 1. Rewrite the import paths locally, either before or after proto generation so that the paths match the local module. This is effectively what we are changing Tendermint to do. 
Option 2. Shove the protos into the `vendor` directory, where protos are allowed to have a name that does not match the module name, but is also a directory that go programmers frequently cede to the `go mod` tooling to manage.

Neither option is great, and I think we can all agree that using the module name `github.com/tendermint/tendermint` in this package is strictly incorrect. If this solution causes serious headache for users who are attempting to vendor these protos into their local `vendor` directory, we can reconsider, but it also seems odd to me that a user would do that and not simply import the generated protos that exist within the Tendermint repository instead since they are effectively equivalent.

Reply:
The issue is that when the user imports the proto file into there proto file and generate their files the go import will be set to something else that isn't actually importing a file. This is how it is currently done for all go projects. 

It's fine if we break this, wanted to make note of it mainly

Reply:
Makes sense, do you know of people doing this?